GREAT INTEGRATION

DSsoto opened the pull request:
Lets make sure that everything is clean here before we merge back to master.

It would be helpful to have as many people as possible review this.

mattlangford committed:
COMMAND: Integrate bounds param into mission runner

mattlangford committed:
TOOLS: Add tool to fudge tf values on the fly

mattlangford committed:
COMMAND: Update bounds system, integrate that system into pose editor

tessbianchi committed:
Merge pull request #73 from mattlangford/master

GPS Bounds and Coordinate Conversion

kev-the-dev committed:
Merge branch 'dev' of https://github.com/uf-mil/Navigator into shooter-dev

Merge uf-mil/dev into shooter-dev

kev-the-dev committed:
Merge uf-mil/dev into dock-shapes-dev

DSsoto committed:
TF: improve tf_fudger tool interface and code structure

Commit:
Merge branch 'master' of https://github.com/uf-mil/Navigator into lidar

DSsoto committed:
Merge pull request #74 from uf-mil/dev

Lake Day 10-2

DSsoto committed:
Merge pull request #41 from ironmig/dock-shapes-dev

PERCEPTION: Add initial dock shape vision service and shooter mission

tessbianchi committed:
Merge branch 'dev' of https://github.com/uf-mil/Navigator into mission_system

Conflicts:
	mission_control/navigator_missions/navigator_singleton/vision_services.yaml
	utils/navigator_msgs/CMakeLists.txt

kev-the-dev committed:
Merged uf-mil/dev to shooter-dev

mattlangford committed:
Merge pull request #75 from tessbianchi/mission_system

Mission system

kev-the-dev committed:
Merge uf-mil/dev to shooter-dev

mattlangford committed:
Merge pull request #71 from ironmig/shooter-dev

SHOOTER: Work with new motor controller + other improvements

mattlangford committed:
TOOLS: Add print function with formatted time

mattlangford committed:
Remove depricated packages.

mattlangford committed:
Merge pull request #77 from ironmig/shooter-dev

SHOOTER: Tuned timing vals/bug fixes

Commit:
Merge branch 'master' of https://github.com/uf-mil/Navigator into lidar

Commit:
Merge branch 'master' of https://github.com/tessbianchi/Navigator into mergeTesting

Commit:
LIDAR: Merge with Tess mission system branch

mattlangford committed:
UTILS: Add rviz waypoint move commands

mattlangford committed:
TOOLS: Add ability to switch from displaying quaternion and euler angles

DSsoto committed:
Merge pull request #79 from mattlangford/master

T-GEN: Add ros interface

mattlangford committed:
Merge pull request #87 from DSsoto/dev

PERCEPTION: ros camera stream class, tf changes, and fix to point cloud colorer

mattlangford committed:
TF: Fix the fixed TF for front stereo

mattlangford committed:
Merge pull request #93 from uf-mil/dev

Lake Day 10-15/16

Commit:
LIDAR: Merge with master

mattlangford committed:
PR: Fix spelling mistakes, remove debug prints, make TF_fudger save to tf.launch

DSsoto committed:
UTILS: add navigator_tools c++ library

DSsoto committed:
PERCEPTION: clean up navigator_perception CMakeLists

DSsoto committed:
PERCEPTION: separate pcd_colorizer node from stereo driver

DSsoto committed:
Merge pull request #88 from mattlangford/master

MISSIONS: Start gate and teleop

mattlangford committed:
Merge pull request #86 from ironmig/detect-deliver-dev-rebased

DETECT DELIVER: Align target w/ lidar & search for color/shape

DSsoto committed:
TOOLS: make print_t() handle printing without rospy initialization

mattlangford committed:
Merge pull request #103 from DSsoto/hydrophones

MISSIONS: Hydrophones

mattlangford committed:
TOOLS: Add 'title' param to print_t

mattlangford committed:
TOOLS: Add color printing features to fprint, transition mission components over to fprint

DSsoto committed:
PERCEPTION: restructure pcd_colorizer and image acquisition code, add pcd IO

Commit:
LIDAR MERGE: Had to add files back that I previously had deleted, will this cause issues?

Commit:
LIDAR: Merge with boat code to add in new lidar changes

mattlangford committed:
Merge pull request #106 from maverick7170/lidar

Updates to Ogrid code

kev-the-dev committed:
PERCEPTION: Rename package to navigator_vison

also fix somethings outputed by DDB -Wall

Commit:
LIDAR: Merge with dev

mattlangford committed:
FPRINT: Add features to fprint and incorporate in mission runner

mattlangford committed:
MISSIONS: Add failure check to move.go()

DSsoto committed:
PERCEPTION: collect all colorizer common resources into 'common.hpp'

DSsoto committed:
Merge pull request #107 from ironmig/detect-deliver-dev

DETECT DELIVER: Track shapes for more reliable perception + colors

DSsoto committed:
Merge pull request #109 from maverick7170/lidar

Lidar: new readme with guide on how to use node

kev-the-dev committed:
DETECT DELIVER: perception services/topics are relative, roi fixed for new cam

kev-the-dev committed:
DETECT DELIVER: Switch to using front camera, resize for performance

kev-the-dev committed:
DETECT DELIVER: Cleaner mission code and fix bug while using front camera

mattlangford committed:
TOOLS: Update tf_fudger to accept '/' in name.

mattlangford committed:
BOUNDS: Add sim mode to bounds server, remove need for convert server to be running for bounds requests

mattlangford committed:
ALARMS: Move station holding to its own node - hopes to fix the station holding issue

mattlangford committed:
OGRID: Add ogrid draw node back, add ogrid fuse script to launch

kev-the-dev committed:
OGRID, MISSIONS, TOOLS: Merge ogrid server, coral surey, start gate, formated debug from mattlangford/master

Added fprint to do some cool fancy printing without 3rd party libraries. You'll notice any mission running or anything that uses the Navigator Singleton will have titles and fancy blue and green text.

move.(...).go() commands will tell you if lqrrt actually got us there or if there was a problem with the trajectory.

Improved start_gate missions and wrote coral_survey mission. coral_survey right now will take rosparam quadrants to search and simply go to the middle of that quadrant (though better search patterns would be pretty trivial). It requires a fake service server to return found shapes right now.

Ogrid arbiter is a thing and can take an arbitrary number of ros occupancy grids and combine them! The ogrids can be arbitrarily sized and positioned but must have the same resolution as the master grid (0.3 right now) and can not be rotated (...yet). That node uses dynamic reconfig so we can make changes to the size of the master ogrid and the topics that it subscribes to on the fly! Things in there are pretty messy math wise and there are rounding errors associated with certain sized master ogrids but due to time I decided to go with it since it works for any case we would throw at it.

Commit:
LIDAR: Merge with dev

mattlangford committed:
Merge pull request #114 from ironmig/detect-deliver-dev

DETECT DELIVER: Lakeday bug fixes and modularity for future missions

mattlangford committed:
GAZEBO: Make absodom correct, meaning bounds can be real now

kev-the-dev committed:
Merge pull request #119 from mattlangford/dev

Lake day 10-30

mattlangford committed:
SIM: Adjust launch files to launch lidar oa in correct order with bounds server

kev-the-dev committed:
LAKE DAY: Merge dev branch changes from lakeday 10-23 and 10-39

Includes tested improvements to tools, detect deliver, ogrid, start gate, and sim

Commit:
LIDAR: Merge with dev

DSsoto committed:
TOOLS: add conversion to sring operator

DSsoto committed:
PERCEPTION: add ros shutdown handling to nodes

tessbianchi committed:
Merge pull request #120 from mattlangford/master

LAUNCH, ALARMS, SIM

tessbianchi committed:
SCAN THE CODE: finishes scan the code mission, updates TF from the boat, updates the mission planner so that it has accounts for a base mission, uses a YAML file and a Queue object

mattlangford committed:
TOOLS: Add script to move fake database objects based on lla coordinates

mattlangford committed:
Merge pull request #123 from maverick7170/lidar

Lidar: Friday Meeting Changes

tessbianchi committed:
Merge branch 'dev' of https://github.com/uf-mil/Navigator into scan_the_code_lidar

Conflicts:
	mission_control/navigator_missions/navigator_singleton/navigator.py
	perception/navigator_lidar_oa/src/ConnectedComponents.h
	perception/navigator_lidar_oa/src/objects.h

tessbianchi committed:
MISSION PLANNER: object database uses incorrect message type

mattlangford committed:
BOUNDS: Move bounds to dynamic reconfig

Commit:
Merge pull request #125 from mattlangford/master

GAZEBO, EXPLORE, OGRID

kev-the-dev committed:
DETECT DELIVER: Clearer text in debug image

kev-the-dev committed:
SIMULATION: Fix sensor pose + tool for getting gazebo pose from tf

Stereo cameras switched to two individual cams in SDF so frame ids are correct

Veloyne points reproject into camera frame correctely in sim

Use ```rosrun navigator_tools tf_to_gazebo.py _cam_fix:=true``` to print out gazebo pose tags based for sensors based on current tf

kev-the-dev committed:
Scan the Code, Mission Planner, Object Classifcation merge from  tessbianchi/scan_the_code_lidar

Merge pull request #124 from tessbianchi/scan_the_code_lidar

mattlangford committed:
OGRID: Draw bounds on ogrid

kev-the-dev committed:
DETECT DELIVER: Better normal approx and debug

Pixel tollerence for normal approx based on shape size

Points included in normal are red in debug image

mattlangford committed:
MOVE: Add circle to move command

tessbianchi committed:
TESTING FRAMEWORK: adds unit test for the mission planner. This also includes ways to spoof services and publishers so we can test independent of other nodes.

kev-the-dev committed:
SHOOTER: Move joystick control to joystick control node

Also minor change in tf gazebo tool to address PR comment

mattlangford committed:
VISION: Add node to estimate color of totem buoys

mattlangford committed:
BOUNDS: Fix bug where node would crash on startup

mattlangford committed:
BOUNDS: Add config files for sim and wauberg to set bounds

Commit:
LIDAR: Merge with dev...

kev-the-dev committed:
DETECT DELIVER: Use PCL for better normal approximation

mattlangford committed:
VISION: Set color in databse

kev-the-dev committed:
TOOLS: Add tool to play video files into ros

kev-the-dev committed:
TOOLS: tf fudger starts with current tf

- Tries for one second to get current tf, otherwise everything is zero

- press r to resetfudge to original

- press z to set everything to zero

tessbianchi committed:
MISSION PLANNER: adds timeouts to missions, constantly checks for object dependencies instead of relying on the missions for this. Adds corresponding tests.

tessbianchi committed:
NAVIGATOR TESTS: removes missing import

mattlangford committed:
COLORAMA: Add service to get color directly

kev-the-dev committed:
TOOLS: add tool to publish voltage and wrench to rviz

Commit:
TESS CHANGES

Conflicts:
	mission_control/navigator_missions/nav_missions/scan_the_code.py

Commit:
TESS CHANGES

tessbianchi committed:
Merge pull request #129 from ironmig/tools-dev

TOOLS: Fudger improvments & video play tool

mattlangford committed:
Merge pull request #138 from whispercoros/remote_control

REMOTE_CONTROL: Controller library and keyboard control

mattlangford committed:
UTILS: Add check for valid rosmsg->numpy conversion

mattlangford committed:
Merge pull request #131 from ironmig/pinger-dev

ACOUSTIC PINGER: Add pinger perception and mission

mattlangford committed:
NAVIGATOR: Make database_query raise an exception if an object isn't found

tessbianchi committed:
MISSION PLANNER: Published when a new missions completes or starts. Fixes some aspects of scan the code, Accepts MissingPerceptionObject as an exception in mission planner

Conflicts:
	mission_control/navigator_missions/nav_missions/scan_the_code.py
	mission_control/navigator_missions/navigator_singleton/navigator.py

tessbianchi committed:
Merge pull request #140 from mattlangford/lake_day_11-13

OGRID, ALARMS, MISSION

kev-the-dev committed:
DETECT DELIVER: Cleaner camera-to-lidar code

May perform better, for sure more readable

also some sneaky fixes in the mission (no yellow\!)

mattlangford committed:
BAG: Add fake action server and image proc

mattlangford committed:
BAG: Add feasibility check to fake action client

Commit:
Merge

mattlangford committed:
FPRINT: Fix missing color, add yellow

Commit:
Merge pull request #143 from uf-mil/dev

LAKE_DAY 11-6 to 11-13

tessbianchi committed:
MISSION PLANNER: adds requested changes to the mission planner

Conflicts:
	perception/navigator_lidar_oa/src/lidar.cpp

tessbianchi committed:
MISSION PLANNER: fix broken dependency on navigator_singleton

tessbianchi committed:
EXPLORE MISSION: adds kwargs to missions, makes explore mission, fixes scan the code to work with new circles

tessbianchi committed:
EXPLORER MISSION: finish the explore mission

DSsoto committed:
PINGER: add least squares based pinger pose estimator

yes I know it's in navigator_vision but just CHILL THE HECK OUT its all good

i also know that david isn't done with this, but it needs to be in repo so we can use without me fumbeling with another screen session

PINGER: add tested least squares pinger

YES I know it is in navigator_vision package. Dont worry it will be fine

we just should have this in the repo and progressvely improve upon it

kev-the-dev committed:
LIDAR: Merge Ira lidar changes into latest dev

tessbianchi committed:
Merge pull request #149 from ironmig/lidar

JUDGES GUI, OBJECT CLASSIFICATION: Ira changes from last few weeks

tessbianchi committed:
LAKE DAY CHANGES: fixes from lake day

kev-the-dev committed:
PINGER: framework for mission with color identification

kev-the-dev committed:
DETECT DELIVER: Fix bug in shape perception when approaching at extreme angles

DSsoto committed:
PINGER: put pinger perception into class, cleaner code

kev-the-dev committed:
TOOLS: add tool to record rviz clicked points to csv

kev-the-dev committed:
TOOLS: Add roi controls to video player

kev-the-dev committed:
DETECT DELIVER, PINGER, TOOLS: address pr comments

mattlangford committed:
Merge pull request #145 from ironmig/detect-deliver-dev

DETECT DELIVER: Lakeday fixes, better alignment

DSsoto committed:
PERCEPTION: refactoring of pcd colorizer that gets observations from all cameras

kev-the-dev committed:
DETECT DELIVER: Catch more pottential errors in camera_to_lidar

mattlangford committed:
Merge pull request #150 from tessbianchi/mission_planner_updates

Mission planner updates

DSsoto committed:
TOOLS: fix bugs and improve usability of video player

DSsoto committed:
MOVE: fix bug with kwargs passed to MoveGoal

Remove kwargs passed to MoveGoal that would crash navc and warn user.
Add support for shorthand commands to navc: 'f', 'b', 'l', 'r', 'yl', 'yr'.
Print more sensible move messages with units.

PR161: address comments

DSsoto committed:
TOOLS: add graceful CTRL-C exit to tf_to_gazebo

mattlangford committed:
COLORAMA: Update how confidence is calucated and updated weights

tessbianchi committed:
PERCEPTION: looking at images for object detection

DSsoto committed:
CORAL SURVEY: add underwater shape templates

DSsoto committed:
PERCEPTION: add underwater shape id node

DSsoto committed:
PERCEPTION: add rotational invariance for filter kernels

DSsoto committed:
PERCEPTION: remove repeated files from bad merge

DSsoto committed:
PERCEPTION: add UnderwaterShapeDetector + Shape class

DSsoto committed:
UTILS: add PI constant to navigator_tools header

DSsoto committed:
PERCEPTION: add config of underwater shape id via yaml

DSsoto committed:
PERCEPTION: make rotateKernel separate and configurable

DSsoto committed:
PERCEPTION: add function to find angle of radial symmetry

tessbianchi committed:
OBJECT CLASSIFICATION: Uses HOG Descriptors to act as a final image object classifier.

tessbianchi committed:
OBJECT CLASSIFICATION: removing unecessary changed for PR

tessbianchi committed:
OBJECT CLASSIFICATION: adds authorship

DSsoto committed:
PERCEPTION: load templates from directory

kev-the-dev committed:
Merge branch 'dev' into devpush

DSsoto committed:
PR172: address comments

kev-the-dev committed:
Merge pull request #160 from zobeld/devpush

REMOTE_CONTROL: Add emergency controller nodes

tessbianchi committed:
OBJECT CLASSIFICATION: changes for PR

tessbianchi committed:
OBJECT CLASSIFICATION: file name change

mattlangford committed:
Merge pull request #174 from uf-mil/dev

Update master branch with pre-competition changes

tessbianchi committed:
EXPLORE_MISSION:
- adds the new explore mission that uses the camera object classification
- changes the roi tool for pause and unpause, instead of having to hold down the space bar

tessbianchi committed:
Merge pull request #162 from mattlangford/dev

NAVC, GAZEBO, START_GATE

tessbianchi committed:
EXPLORE_MISSION:
- creates an explore mission that confirms the identity of the object before beginning the mission, and
locks that object into the database

kev-the-dev committed:
IDENTIFY DOCK: untested mission for docking using dd vision service

has two mission options, one relying on a perception node that hasn't been written to identify the location of bays, and another simply moving in front of symbols like in detect deliver, assuming they are in the center

dock progress rebase

docking bug fixes

DSsoto committed:
PERCEPTION: find all perturbations of a curve segment

kev-the-dev committed:
PINGER: exit through gate after circling

docking last minute

yaml

tessbianchi committed:
EXPLORE MISSION: updates for pr

mattlangford committed:
BOUNDS: Add gps coordinates for the channel we compete in

mattlangford committed:
Merge pull request #178 from tessbianchi/explore_mission

Explore mission

mattlangford committed:
COLORAMA: Add debug message

DSsoto committed:
PERCEPTION: add utility to sort curves by length

kev-the-dev committed:
PERCEPTION: Improve normal approx in shapefinder

mattlangford committed:
Merge pull request #182 from ironmig/competition-dev

PINGER, DOCKING, DETECT DELIVER, SHOOTER, PERCEPTION, TOOLS: kevins work from compeition through 12/14

DSsoto committed:
PERCEPTION: perturb closed curves by index

DSsoto committed:
PINGER: correct 'None' return of service

DSsoto committed:
USID: test active contours

mattlangford committed:
BOUNDS: Add updated bounds files

tessbianchi committed:
scan_the_code

mattlangford committed:
Merge pull request #186 from ironmig/competition-dev-with-david

COMPETITION: Mission and other changes from Kevin + David

tessbianchi committed:
scan the code

tessbianchi committed:
changes to stc, and db

tessbianchi committed:
changed mission planner to allow for sime mode

forrestv committed:
DETECT DELIVER: add perception node to track shooter platform

mattlangford committed:
Merge pull request #188 from ironmig/competition-dev

MISSIONS: Saturday fixes to pinger, detect deliver, identify dock

Commit:
Merge pull request #191 from uf-mil/dev

Final master dev merge

mattlangford committed:
TOOLS: Add fprint, add more generic tool interface

mattlangford committed:
MISSIONS: Add vision proxy config, port some more navigator files back

mattlangford committed:
FPRINT: Glorious updates

DSsoto committed:
SONAR: fix sporadic bootloading bug

RustyBamboo committed:
FPRINT: Fixed typo in initiating FPrintFactor with default color

DSsoto committed:
Merge pull request #189 from mattlangford/master

ALARMS: Overhaul

DSsoto committed:
TESTS: add ros_alarms server to integration test

mattlangford committed:
FPRINT: Add auto_bold option

DSsoto committed:
Merge pull request #192 from mattlangford/master

ALARMS: Updated for new txros alarms

DSsoto committed:
PERCEPTION: move sub pkg to top level perception dir

sentree committed:
INSTALL: Replace all progressbar dependencies with tqdm

DSsoto committed:
Merge pull request #192 from whispercoros/master

REPO: Clean up and adjustment to work with software-common

sentree committed:
REPO: Update references to uf-mil repositories after renaming

Commit:
UTILS: Add online_bagger

Commit:
ONLINE_BAGGER: Add resubscribe on failed topics to callback

DSsoto committed:
Merge pull request #193 from whispercoros/master

REPO: Update references to uf-mil repositories after renaming

sentree committed:
Merge branch 'master' of SubjuGator into great_merge

sentree committed:
Merge branch 'master' of NaviGator into great_merge

DSsoto committed:
Merge pull request #11 from whispercoros/great_merge

REPO: The Great Merge (Reprise)

kev-the-dev committed:
REPO: temporarily move all great merge files to .great_merge so catkin doesnt build it

kev-the-dev committed:
REPO: move sub8 tools to mil_dev_tools

need to refactor later

kev-the-dev committed:
REPO: clear remaining sub8 tools

kev-the-dev committed:
REPO: move navigator_tools files to mil_dev_tools, purge remaining

kev-the-dev committed:
REPO: merge overlapped tools files in sub8 and navigator

kev-the-dev committed:
TOOLS: change naming to mil_dev_tools in imported tools

kev-the-dev committed:
TOOLS: fix import error in geometry helpers

kev-the-dev committed:
TOOLS: fix typo in msg helpers

DSsoto committed:
GREAT MERGE: rename navigator_vision to mil_vision

navigator_vision is moved from the hidden '.great_merge' directory to
the top-level 'perception' directory. navigator_vision was a superset
of sub8_perception, 2 tools need to be added and vehicle particular
code should be removed.

DSsoto committed:
GREAT MERGE: remove files that were specific to a robot

DSsoto committed:
GREAT MERGE: create purgatory for files that aren't yet generic

DSsoto committed:
GREAT MERGE: change navigator_vision refs to mil_vision

DSsoto committed:
GREAT MERGE: transform uf_common to mil_msgs

DSsoto committed:
GREAT MERGE: convert sub8_sonar to mil_passive_sonar

Added the passive to the name because we now have an imaging sonar.

DSsoto committed:
GREAT MERGE: convert sub8_sonar to mil_passive_sonar

Added 'passive' to the name because we now have an imaging sonar.

DSsoto committed:
SONAR: add script to generate dtoas to test hardware

DSsoto committed:
GREAT MERGE: adapt mil_vision to compile w/ new names

DSsoto committed:
GREAT MERGE: add mil_tools cpp lib

DSsoto committed:
GREAT MERGE: add sub8_perception tools to mil_vision

kev-the-dev committed:
TOOLS: rename mil_dev_tools to mil_tools for ease

more sensible to have tools package as it already contains non-development tools

kev-the-dev committed:
REPO: add CATKIN_IGNORE to greatmerge temp dir

kev-the-dev committed:
REPO: move mil_tools to utils/

kev-the-dev committed:
VISION: add mil_tools as dependency to vision

DSsoto committed:
Merge pull request #12 from ironmig/sweet-great-sweet

TOOLS: import tools from navigator and sub8

Commit:
ONLINE BAGGER: Add ram limit

Commit:
ONLINE_BAGGER: Enforce PEP8

Commit:
ONLINE_BAGGER: Add custom service BaggerCommands.srv

Commit:
ONLINE_BAGGER: Remove Client Example

Commit:
ONLINE_BAGGER: Fix failed topic subscription

Commit:
ONLINE_BAGGER: Add bag naming feature to client

Commit:
ONLINE_BAGGER: Fix variable naming conventions

Commit:
ONLINE_BAGGER: Add individual bag times to topic config

Commit:
ONLINE_BAGGER: Add bag time to service call

DSsoto committed:
Merge pull request #6 from kingkevlar/online_bagger

UTILS: Add online_bagger

DSsoto committed:
PASSSIVE SONAR: remove legacy hydrophones pkg

The ros package mil_passive_sonar should now be used for locating pingers.

The old hydrophone package was only able to work under one hydrophone
configuration. It should be removed from the SubjuGator repo.

DSsoto committed:
PASSIVE SONAR: move paulboard pkg to mil_passive_sonar

DSsoto committed:
PASSIVE SONAR: make paulboard_driver a python pkg

The scripts here were demoted from being a ros package. It should be
removed from the SubjuGator repo.

DSsoto committed:
PASSIVE SONAR: move pinger finder node to mil_passive_sonar

It was previously in the great_merge_purgatory of mil_vision

DSsoto committed:
VISION: integrate camera_lidar_transformer

Node used to be in the great_merge_transformer subdirectory.
All other nodes were considered not generic and removed from the repo.

kev-the-dev committed:
VISION: copy CameraToLidarTransform.srv from NaviGator

DSsoto committed:
VISION: move camera_stream_demo.cpp from nodes to src

It was decided that all the nodes directory would only be used for python nodes, wile the src and include directories will include c++ nodes, not just library code.

kev-the-dev committed:
TOOLS: add script to rename topics/frames in bags

DSsoto committed:
Merge pull request #21 from ironmig/fix-bag

TOOLS: add script to rename topics/frames in bags

RustyBamboo commented on perception/mil_vision/great_merge_purgatory/README.txt:
```diff
@@ -0,0 +1,5 @@
+Some of these files can be made generic and therefore be able to stay as a part of mil_common. Until that is done, they will remain in this folder.
```
great_merge_purgatory

kev-the-dev commented on perception/mil_vision/include/mil_vision_lib/colorizer/camera_observer.hpp:
```diff
@@ -0,0 +1,45 @@
+#include <image_geometry/pinhole_camera_model.h>
+#include <opencv2/core/eigen.hpp>
+
+namespace nav{
```
All namespaces should be ```mil_vision``` for resources in this library.

kev-the-dev commented on perception/mil_vision/include/mil_vision_lib/colorizer/camera_observer.hpp:
```diff
@@ -0,0 +1,45 @@
+
+namespace nav{
+  
+class CameraObserver
```
Let's have just one (David's) wrapper for camera subscribers/publishers. @DSsoto can you remove the other ones that do the same thing?

kev-the-dev commented on perception/mil_vision/include/mil_vision_lib/cv_tools.hpp:
```diff
@@ -0,0 +1,167 @@
+                                    const float low_thresh_gain = 0.5,
+                                    const float high_thresh_gain = 0.5);
+
+cv::Mat triangulate_Linear_LS(cv::Mat mat_P_l, cv::Mat mat_P_r,
```
No documentation for this function. I do I use it/ what does it do? Same with others

kev-the-dev commented on perception/mil_vision/include/mil_vision_lib/point_cloud_algorithms.hpp:
```diff
@@ -0,0 +1,70 @@
+};
+
+
+class PcdColorizer : public PcdSubPubAlgorithm{
```
Add TODO comment, mention unfinished

kev-the-dev commented on perception/mil_vision/include/mil_vision_lib/visualization.hpp:
```diff
@@ -0,0 +1,24 @@
+  ros::Publisher rviz_pub;
+  ros::NodeHandle nh;
+  RvizVisualizer(std::string rviz_topic);
+  void visualize_buoy(geometry_msgs::Pose &pose, std::string &frame_id);
```
sub specific, remove

kev-the-dev commented on perception/mil_vision/object_classification/lidar_to_image.py:
```diff
@@ -0,0 +1,194 @@
+___author___ = "Tess Bianchi"
+
+
+class LidarToImage(object):
```
Specific to Navigator, refactor or remove

kev-the-dev commented on perception/mil_vision/object_classification/object_classification.py:
```diff
@@ -0,0 +1,150 @@
+
+from roi_generator import ROI_Collection
+from navigator_tools import CvDebug, BagCrawler
```
Navigator dependency, update to mil_common 

Specific to Navigator, refactor or remove

kev-the-dev commented on perception/mil_vision/object_classification/roi_generator.py:
```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/python
+from navigator_tools import BagCrawler
```
object_classification folder should be removed or refactored to be generic

kev-the-dev commented on perception/mil_vision/package.xml:
```diff
@@ -0,0 +1,40 @@
+  <build_depend>tf2_sensor_msgs</build_depend>
+  <run_depend>tf2_sensor_msgs</run_depend>
+  <build_depend>tf2_geometry_msgs</build_depend>
+  <build_depend>navigator_tools</build_depend>
```
mil_common should not depened on navgiator

kev-the-dev commented on perception/mil_vision/ros_tools/easy_thresh.py:
```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/python
```
Broken for me on kinetic, someone else try to run it

kev-the-dev commented on perception/mil_vision/ros_tools/on_the_fly_thresholder.py:
```diff
@@ -0,0 +1,117 @@
+from sensor_msgs.msg import Image, CompressedImage
+from cv_bridge import CvBridge, CvBridgeError
+
+from navigator_tools import image_helpers
```
use mil_common instead

kev-the-dev commented on perception/mil_vision/setup.py:
```diff
@@ -0,0 +1,11 @@
+
+# fetch values from package.xml
+setup_args = generate_distutils_setup(
+    packages=[''],
```
Should this be empty? Can you import the python stuff from this package?

kev-the-dev commented on perception/mil_vision/src/mil_vision_lib/active_contours.cpp:
```diff
@@ -0,0 +1,253 @@
+namespace nav
+{
+
+namespace Perturbations
```
Documentation

kev-the-dev commented on perception/mil_vision/src/mil_vision_lib/colorizer/camera_observer.cpp:
```diff
@@ -0,0 +1,73 @@
+
+using mil::tools::operator "" _s; // converts to std::string
+  
+CameraObserver::CameraObserver(ros::NodeHandle &nh, std::string &pcd_in_topic, std::string &cam_topic, size_t hist_size)
```
Document. What does this do?

kev-the-dev commented on perception/mil_vision/src/mil_vision_lib/colorizer/color_observation.cpp:
```diff
@@ -0,0 +1,15 @@
+#include <colorizer/color_observation.hpp>
```
Document the colorizer package. Add a readme

kev-the-dev commented on perception/mil_vision/src/mil_vision_lib/colorizer/pcd_colorizer.cpp:
```diff
@@ -0,0 +1,173 @@
+using namespace std;
+using namespace cv;
+
+namespace nav{
```
namespace

kev-the-dev commented on perception/mil_vision/src/mil_vision_lib/cv_utils.cc:
```diff
@@ -0,0 +1,656 @@
+#include <mil_vision_lib/cv_tools.hpp>
```
Rename to cpp

Document these functions.

kev-the-dev commented on utils/mil_msgs/CMakeLists.txt:
```diff
@@ -0,0 +1,43 @@
+  MoveTo.action
+)
+
+add_message_files(FILES
```
Some of these exist in geometry_msgs/other core ros packages.

kev-the-dev commented on utils/mil_msgs/CMakeLists.txt:
```diff
@@ -0,0 +1,43 @@
+catkin_python_setup()
+
+add_action_files(FILES
+  MoveTo.action
```
Is this vehicle dependent?

kev-the-dev commented on utils/mil_msgs/include/mil_msgs/msg_helpers.h:
```diff
@@ -0,0 +1,77 @@
+#include <Eigen/Dense>
+#include <tf/tf.h>
+
+namespace uf_common {
```
namespace

kev-the-dev commented on utils/mil_msgs/include/mil_msgs/msg_helpers.h:
```diff
@@ -0,0 +1,77 @@
+#pragma once
```
msg_helpers should be moved to dev tools package. This package should be just message/service/action definitions common to mil projects

kev-the-dev commented on utils/mil_msgs/scripts/param_saver:
```diff
@@ -0,0 +1,40 @@
+import yaml
+
+import roslib
+roslib.load_manifest('uf_common')
```
don't need with catkin. Also these scripts should go to tools package

kev-the-dev commented on utils/mil_tools/include/mil_tools/mil_tools.hpp:
```diff
@@ -0,0 +1,33 @@
+#include <opencv2/core/core.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+
+namespace mil{
```
rename to mil_tools

kev-the-dev commented on utils/mil_tools/mil_misc_tools/download.py:
```diff
@@ -0,0 +1,57 @@
+
+
+if __name__ == '__main__':
+    sub_model_url = "http://goo.gl/f0ennf?gdriveurl"
```
bad

kev-the-dev commented on utils/mil_tools/mil_ros_tools/bag_crawler.py:
```diff
@@ -0,0 +1,71 @@
+import tqdm
+
+
+class BagCrawler(object):
```
Document

kev-the-dev commented on utils/mil_tools/mil_ros_tools/func_helpers.py:
```diff
@@ -0,0 +1,38 @@
+    print add(2, 3)
+    print add(2, 3)
+
+    print gooberstein("hello")
```
wat

kev-the-dev commented on utils/mil_tools/mil_ros_tools/func_helpers.py:
```diff
@@ -0,0 +1,38 @@
+
+
+@Cache
+def add(a, b):
```
useful

kev-the-dev commented on utils/mil_tools/mil_ros_tools/image_helpers.py:
```diff
@@ -0,0 +1,116 @@
+from mil_ros_tools import wait_for_param
+
+
+def get_parameter_range(parameter_root):
```
sub specific, remove

kev-the-dev commented on utils/mil_tools/mil_ros_tools/image_helpers.py:
```diff
@@ -0,0 +1,116 @@
+        self.bridge = CvBridge()
+        self.callback = callback
+
+    def wait_for_camera_info(self, timeout=10):
```
known bug when time goes backwards, fixed on sub8 version

kev-the-dev commented on utils/mil_tools/mil_ros_tools/msg_helpers.py:
```diff
@@ -0,0 +1,194 @@
+import rospy
+
+
+def rosmsg_to_numpy(rosmsg, keys=None):
```
@RustyBamboo has some fix for this. Please PR it

kev-the-dev commented on utils/mil_tools/nodes/tf_fudger.py:
```diff
@@ -0,0 +1,172 @@
+    if k == ord('z'):
+        set_bars((0, 0, 0), (0, 0, 0))
+
+    # This functionality was configured to replace lines in navigator's tf launch file, should refactor to be general later
```
IF someone wants to refactor to be generic, that'd be cool

kev-the-dev commented on utils/mil_tools/nodes/video_player:
```diff
@@ -0,0 +1,134 @@
+      self.paused = not self.paused
+
+def main():
+    rospy.init_node('video_player', anonymous=True)
```
use argparse instead of params

kev-the-dev reviewed (changes requested):
Could use some work. 6/10

Some of these changes need to be done before merging:

* namespace changes
* anything specific to sub8/navigator

Later
* document

DSsoto committed:
GREAT MERGE: make DepthStamped.msg (mil_msgs)

This is new mesage should be used by the depth driver instead of
Float64Stamped

DSsoto committed:
GREAT MERGE: deprecate Float64Stamped.msg

Replaced with RangeStamped. DVL related cold that previously used
Float64Stamped should be transitioned to use RangeStamped instead.

Other depth driver related code should be changed to use
DepthStamped.

The only usage of Float64Stamped that is not DVL or depth or depth
driver related is in the start gate mission and should be removed
(@RustyBamboo).

kev-the-dev committed:
REPO: various changes for sub8 integration

* fix up namespaces in mil_vision

* move mil_msgs stuff that isnt messages to mil_tools

kev-the-dev committed:
REPO: remove sub8 specific visualizaton

DSsoto committed:
GREAT MERGE: include new msgs in mil_msgs cmkaelists

sentree committed:
Merge pull request #25 from DSsoto/deprecate_float64stamped

Remove Float64stamped

DSsoto committed:
GREAT MERGE: add camera_lidar_transform to mil_msgs cmake

kev-the-dev committed:
Merge pull request #26 from DSsoto/great_merge

Remove mil_vision/great_merge_purgatory

kev-the-dev committed:
DEPENDENCIES: update odometry_utils submodule

sentree committed:
Merge pull request #28 from ironmig/great_merge

REPO: Namespace changes to prepare for sub8 integration

kev-the-dev committed:
REPO: replace navigator dependencies with mil_common

sentree committed:
Merge pull request #29 from ironmig/great_merge

REPO: replace navigator dependencies with mil_common
